File: src/06.hashMap/04.128.ts
export function longestConsecutive(nums: number[]): number {
    const set = new Set<number>();

    for (const num of nums) {
        set.add(num);
    }

    let longest = 0;

    for (const num of set) {
        if (!set.has(num - 1)) {
            let curNum = num;
            let curLong = 1;

            while (set.has(curNum + 1)) {
                curNum += 1;
                curLong += 1;
            }

            longest = Math.max(longest, curLong);
        }
    }

    return longest;
}

File: src/06.hashMap/04.128.go
package hashmap

func LongestConsecutive(nums []int) int {
	hashMap := map[int]bool{}
	for _, v := range nums {
		hashMap[v] = true
	}

	longest := 0

	for k := range hashMap {
		if _, ok := hashMap[k-1]; !ok {
			curNum := k
			curLong := 1

			for hashMap[curNum+1] {
				curNum += 1
				curLong += 1
			}

			if longest < curLong {
				longest = curLong
			}
		}
	}

	return longest
}
